src/sentry/integrations/bitbucket_server/repository.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import absolute_import
+
+import six
+
+from django.core.urlresolvers import reverse
+from sentry.models.integration import Integration
+from sentry.plugins.providers.integration_repository import IntegrationRepositoryProvider
+from sentry.utils.http import absolute_uri
+from sentry.integrations.exceptions import ApiError, IntegrationError
+
+
+class BitbucketServerRepositoryProvider(IntegrationRepositoryProvider):
+    name = "Bitbucket Server"
+
+    def get_installation(self, integration_id, organization_id):
+        if integration_id is None:
+            raise IntegrationError("Bitbucket Server requires an integration id.")
+        integration_model = Integration.objects.get(
+            id=integration_id, organizations=organization_id, provider="bitbucket_server"
+        )
+
+        return integration_model.get_installation(organization_id)
+
+    def get_repository_data(self, organization, config):
+        installation = self.get_installation(config.get("installation"), organization.id)
+        client = installation.get_client()
+        try:
+            parts = config["identifier"].split("/")
+            repo = client.get_repo(parts[0], parts[1])
+        except Exception as e:
+            installation.raise_error(e)
+        else:
+            config["external_id"] = six.text_type(repo["id"])
+            config["name"] = repo["project"]["name"] + "/" + repo["name"]
```

Comment:
@pierredup I see that we are often splitting the name downstream so we can split the project and repo name. This makes it a bit confusing because it might be super clear that `parts[0]` is the project and `parts[1]` is the name in other places in the code. Any reason we can't store the `project` so the name isn't `{project}/{name}`? We could call the latter field `long_name` if we still need it in other places.